src/backend/columnar/columnar_metadata.c

```diff
@@ -218,6 +219,117 @@ InitColumnarOptions(Oid regclass)
 }
 
 
+/*
+ * ExtractColumnarOptions - split inoutOptions into two lists, one for
+ * columnar options and another for everything else.
+ */
+List *
+ExtractColumnarRelOptions(List *inOptions, List **outColumnarOptions)
+{
+	List *columnarOptions = NIL;
+	List *otherOptions = NIL;
+	ListCell *lc;
+
+	foreach (lc, inOptions)
+	{
+		DefElem *elem = castNode(DefElem, lfirst(lc));
+
+		if (elem->defnamespace != NULL &&
+			strcmp(elem->defnamespace, COLUMNAR_RELOPTION_NAMESPACE) == 0)
+		{
+			columnarOptions = lappend(columnarOptions, elem);
+		}
+		else
+		{
+			otherOptions = lappend(otherOptions, elem);
+		}
+	}
+
+	*outColumnarOptions = columnarOptions;
+
+	return otherOptions;
+}
+
+
+/*
+ * SetColumnarRelOptions - apply the list of DefElem options to the relation.
+ */
+void
+SetColumnarRelOptions(RangeVar *rv, List *reloptions)
+{
+	ColumnarOptions options;
+
+	if (reloptions == NIL)
+	{
+		return;
+	}
+
+	Relation rel = relation_openrv(rv, AccessShareLock);
+	Oid relid = RelationGetRelid(rel);
+	relation_close(rel, NoLock);
+
+	/* get existing or default options */
+	if (!ReadColumnarOptions(relid, &options))
+	{
+		/* if extension doesn't exist, just return */
+		return;
+	}
+
+	ListCell *lc;
+	foreach (lc, reloptions)
+	{
+		DefElem *elem = castNode(DefElem, lfirst(lc));
+
+		if (elem->defnamespace == NULL ||
+			strcmp(elem->defnamespace, COLUMNAR_RELOPTION_NAMESPACE) != 0)
+		{
+			ereport(ERROR, (errmsg("columnar options must have the prefix \"%s\"",
+								   COLUMNAR_RELOPTION_NAMESPACE)));
+		}
+
+		if (strcmp(elem->defname, "chunk_group_row_limit") == 0)
+		{
+			options.chunkRowCount = (elem->arg == NULL) ?
+				columnar_chunk_group_row_limit : defGetInt64(elem);
+		}
+		else if (strcmp(elem->defname, "stripe_row_limit") == 0)
+		{
+			options.stripeRowCount = (elem->arg == NULL) ?
+				columnar_stripe_row_limit : defGetInt64(elem);
+		}
+		else if (strcmp(elem->defname, "compression") == 0)
+		{
+			if (elem->arg == NULL)
+			{
+				options.compressionType = columnar_compression;
+			}
+			else
+			{
+				char *compressionStr = defGetString(elem);
+				CompressionType compressionType = ParseCompressionType(compressionStr);
+
+				if (options.compressionType == COMPRESSION_TYPE_INVALID)
+				{
+					ereport(ERROR, (errmsg("invalid columnar compression type: \"%s\"",
+										   compressionStr)));
+				}
+				else
+				{
+					options.compressionType = compressionType;
+				}
+			}
+		}
+		else if (strcmp(elem->defname, "compression_level") == 0)
+		{
+			options.compressionLevel = (elem->arg == NULL) ?
+				columnar_compression_level : defGetInt64(elem);
+		}
```

Comment:
I was expecting a warning or error here if an unkown columnar option would be tried to set. Do we sanitize the option list elsewhere? Or is there an other good reason to allow arbitrary `columnar.options` to be set?

Reply:
As expected the current behaviour doesn't error/warn when we set a non-existing columnar option. Instead it is silently ignored:

```
[local] citus@citus:9700-18928=# ALTER TABLE c SET (columnar.barbaz = fooz);
ALTER TABLE
Time: 76.822 ms
```

Would this be desired behaviour?

default behaviour of heap is:
```
[local] citus@citus:9700-18928=# ALTER TABLE t1 SET (something_bogus = 42);
ERROR:  unrecognized parameter "something_bogus"
Time: 0.540 ms
```

I guess we would want to retain such behaviour.

Reply:
Fixed.